daft/iceberg/iceberg_write.py

```diff
@@ -222,13 +222,13 @@ def partitioned_table_to_iceberg_iter(
     partition_values = partitioned.partition_values()
 
     if partition_values:
-        partition_strings = partition_values_to_string(partition_values, partition_null_fallback="null").to_pylist()
+        partition_strings = partitioned.partition_values_str().to_pylist()  # type: ignore
```

Comment:
Why is `type: ignore` necessary here?

Reply:
`partition_values_str` may return None which mypy doesn't like. However we know that at this point it should return a micropartition

Reply:
You can add an `assert is not None` here instead, which is more explicit and *should also make mypy behave